### PR TITLE
Use a default user agent 'kuberay-operator' instead of the default user-agent from controller-runtime

### PR DESCRIPTION
**Pull request by andrewsykim**
## Why are these changes needed?

We are currently using the default controller-runtime user-agent which is `manager/v0.0.0`. This PR updates Kuberay's user-agent to `kuberay-operator`. 

Note that it does not include the version of kuberay yet, but we should add that in a follow-up PR. Here's example HTTP log from apiserver that shows the new user-agent being used:
```
I0311 02:35:26.041556       1 httplog.go:131] "HTTP" verb="PUT" URI="/apis/ray.io/v1/namespaces/default/rayservices/rayservice-sample/status" latency="10.397049ms" userAgent="kuberay-operator" audit-ID="8b543d30-4d26-4e0f-96ff-7b2596d0ecb8" srcIP="10.244.0.3:54042" apf_pl="workload-low" apf_fs="service-accounts" apf_iseats=1 apf_fseats=0 apf_execution_time="8.0205ms" resp=200
```

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by andrewsykim** [comment]
@kevin85421 PTAL

**Review by kevin85421 — COMMENTED** [review]
Would you mind sharing more details on why we need this change? I don't have the related context. Thanks!

**Comment by andrewsykim** [comment]
> Would you mind sharing more details on why we need this change? I don't have the related context. Thanks!

It's mainly for auditing HTTP traffic in kube-apiserver. If we use the default user-agent from controller-runtime (`manager/v0.0.0`), it will be hard to distinguish kuberay traffic from other clients that also use controller-runtime. Changing this to `ray-operator` will make it easy to filter out HTTP logs that came from kuberay

File: ray-operator/main.go [hunk 1]
```diff
@@ -42,6 +42,8 @@ var (
 	_commitId_  = ""
 	scheme      = runtime.NewScheme()
 	setupLog    = ctrl.Log.WithName("setup")
+	// TODO: include the kuberay version in the user-agent. E.g. ray-operator/v1.1.0
+	userAgent = "ray-operator"
```
Comment (kevin85421) [inline ray-operator/main.go:46]:
```suggestion
	userAgent = "kuberay-operator"
```

**Review by kevin85421 — APPROVED** [review]
How do I manually test this PR? I checked the apiserver logs in my local Kind cluster by running `kubectl logs -n kube-system kube-apiserver-kind-control-plane`, but I didn't find any log message similar to `[...httplog.go:131] "HTTP" verb="PUT" ....`.

**Comment by kevin85421** [comment]
cc @Yicheng-Lu-llll would you mind reviewing this PR? Thanks!

**Comment by Yicheng-Lu-llll** [comment]
lgtm! I can reproduce it.
image

**Review by Yicheng-Lu-llll — APPROVED** [review]

**Pull request merged**